js/index.js

```diff
@@ -142,36 +233,68 @@ Server.prototype = {
         ng2: this.pluginLoader.ng2
       },
       newPluginHandler: (pluginDef) => this.newPluginSubmitted(pluginDef),
-      auth: webauth,
+      auth: webauth
     };
     this.webApp = makeWebApp(webAppOptions);
     this.webServer.startListening(this.webApp.expressApp);
-    this.pluginLoader.on('pluginAdded', event => this.pluginLoaded(event.data));
+    const eurekaJSON = this.userConfig.node.mediationLayer;
+
+    // Add to eureka metadata info
+
+    var totalPlugins = -1;    // Stores the number of plugins that should load
+    var addedPlugins = 0;     // Stores the number of plugins that have attempted to load
+    
+    // Get number of plugins that should load
+    this.pluginLoader.on('givePluginAmount', event => totalPlugins = event.data);
+    
+    // Add the plugin and add it to the count
+    this.pluginLoader.on('pluginAdded', event => {
+      this.pluginLoaded(event.data);
+      addedPlugins++;            // Make sure total plugins did not get overwritten to null
+
+      if (totalPlugins != null && totalPlugins == addedPlugins) {   // If all the plugins have attempted to load
```

Comment:
What happens when a dynamic plugin is added? Do we care if it's registered to Eureka or not?

Reply:
I don't think the mediation layer currently supports this, but it would be good.